backend/app/services/transcription.py

```
@@ -0,0 +1,283 @@
+        return segment
+
+    async def _transcribe_with_retry(self, pcm16_audio: bytes) -> tuple[str, float, int]:
+        max_attempts = 4
+        base_delay = 0.4
+        last_error: Exception | None = None
+
+        for attempt in range(max_attempts):
+            try:
+                text, confidence = await self._transcribe_openai(pcm16_audio)
+                return text, confidence, attempt
+            except Exception as openai_err:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)